paper-api/src/main/java/org/bukkit/entity/Player.java

```diff
@@ -3534,24 +3534,23 @@ default net.kyori.adventure.text.event.HoverEvent<net.kyori.adventure.text.event
     <T> T getClientOption(com.destroystokyo.paper.ClientOption<T> option);
     // Paper end - client option API
 
-    // Paper start - elytra boost API
     /**
      * Boost a Player that's {@link #isGliding()} using a {@link Firework}.
      * If the creation of the entity is cancelled, no boosting is done.
      * This method does not fire {@link com.destroystokyo.paper.event.player.PlayerElytraBoostEvent}.
      *
      * @param firework The {@link Material#FIREWORK_ROCKET} to boost the player with
      * @return The {@link Firework} boosting the Player or null if the spawning of the entity was cancelled
-     * @throws IllegalArgumentException if {@link #isGliding()} is false
-     * or if the {@code firework} isn't a {@link Material#FIREWORK_ROCKET}
+     * @throws IllegalArgumentException if the {@code firework} isn't a {@link Material#FIREWORK_ROCKET}
+     * @throws IllegalStateException if {@link #isGliding()} is false
      * @deprecated use {@link HumanEntity#fireworkBoost(ItemStack)} instead. Note that this method <b>does not</b>
      * check if the player is gliding or not.
      */
```

Comment:
These comments need to be updated as it can now be a non firework item it seem?

Reply:
True i remove this now.

Reply:
This also include the parameter description.